scripts/formatter.sh

```
@@ -0,0 +1,17 @@
+set -euo pipefail
+
+if [ "${1:-}" = "--check" ]; then
+  MODE=(--dry-run --Werror)
+else
+  MODE=(-i)
+fi
+
+find \
+    src \
+    -iname "*.h" -o -iname "*.hpp" -o -iname "*.cpp" -o -iname "*.c" \
+  -print0 | xargs -0 clang-format-14 "${MODE[@]}"
```

Shellcheck (0.11.0)

[warning] This action ignores everything before the -o. Use \( \) to group.

(SC2146)